usaspending_api/search/v2/views/search_elasticsearch.py

```diff
@@ -49,6 +53,16 @@ def post(self, request):
                 m["optional"] = False
         validated_payload = TinyShield(models).block(request.data)
 
+        record_num = (validated_payload["page"] - 1) * validated_payload["limit"]
+        if record_num >= settings.ES_TRANSACTIONS_MAX_RESULT_WINDOW:
+            raise UnprocessableEntityException(
+                "Accessing page {page} with limit {limit} exceeds the Elasticsearch limit of {es_limit}. Please use downloads to access the full set of results.".format(
```

Comment:
It's a good practice to hide backend architecture from the API. Something along these lines?

`Page #{page} of size {limit} is over the maximum result limit ({es_limit}). Consider using custom data downloads to obtain large data sets.

Reply:
changed.